Gems/Atom/Feature/Common/Editor/Scripts/ColorGrading/lut_helper.py

```diff
@@ -0,0 +1,233 @@
+
+# Set the destination image pixels by applying the shaperfunction
+for y in range(outBuf.ybegin, outBuf.yend):
+    for x in range(outBuf.xbegin, outBuf.xend):
+        srcPx = buf.getpixel(x, y)
+        # _LOGGER.debug(f'srcPx is: {srcPx}')
+        if op == 0:
+            dstPx = (ShaperInv(bias, scale, srcPx[0]), ShaperInv(bias, scale, srcPx[1]), ShaperInv(bias, scale, srcPx[2]))
+            outBuf.setpixel(x, y, dstPx)
+        elif op == 1:
+            dstPx = (Shaper(bias, scale, srcPx[0]), Shaper(bias, scale, srcPx[1]), Shaper(bias, scale, srcPx[2]))
+            outBuf.setpixel(x,y, dstPx)
+        else:
+            # Unspecified operation. Just write zeroes
+            outBuf.setpixel(x, y, 0.0, 0.0, 0.0)
+_LOGGER.info("writing %s..." % (args.o))
+outBuf.write(args.o, "float")
+
+lutIntervals = []
+lutValues = []
+if args.write3dl or args.writeAsset:
+    # First line contains the vertex intervals
+    dv = 1023.0 / float(lutSize-1)
+    for i in range(lutSize):
+        lutIntervals.append(np.uint64(dv * i))
+    # Texels are in R G B per line with indices increasing first with blue, then green, and then red.
+    for r in range(lutSize):
+        for g in range(lutSize):
+            for b in range(lutSize):
+                uv = GetUvCoord(lutSize, r, g, b)
+                px = outBuf.getpixel(uv[0], uv[1])
+                lutValues.append((np.uint64(px[0] * 4095), np.uint64(px[1] * 4095), np.uint64(px[2] * 4095)))
+
+if args.write3dl:
+    lutFileName = "%s.3dl" % (args.o)
+    _LOGGER.info("writing %s..." % (lutFileName))
+    lutFile = open(lutFileName, 'w')
+    # First line contains the vertex intervals
+    dv = 1023.0 / float(lutSize)
+    for i in range(lutSize):
+        lutFile.write("%d " % (lutIntervals[i]))
+    lutFile.write("\n")
+    for px in lutValues:
+        lutFile.write("%d %d %d\n" % (px[0], px[1], px[2]))
+    lutFile.close()
+
+if args.writeAsset:
+    assetFileName = "%s.azasset" % (args.o)
+    _LOGGER.info("writing %s...", (assetFileName))
+    assetFile = open(assetFileName, 'w')
+    assetFile.write("{\n")
+    assetFile.write("    \"Type\": \"JsonSerialization\",\n")
+    assetFile.write("    \"Version\": 1,\n")
+    assetFile.write("    \"ClassName\": \"LookupTableAsset\",\n")
+    assetFile.write("    \"ClassData\": {\n")
+    assetFile.write("        \"Name\": \"LookupTable\",\n")
+    assetFile.write("        \"Intervals\": [\n")
+    for i in range(lutSize):
+        assetFile.write(" %d" % (lutIntervals[i]))
+        if i < (lutSize-1):
+            assetFile.write(", ")
+    assetFile.write("],\n")
+    assetFile.write("        \"Values\": [\n")
+    for idx, px in enumerate(lutValues):
+        assetFile.write(" %d, %d, %d" % (px[0], px[1], px[2]))
+        if idx < (len(lutValues) - 1):
+            assetFile.write(",")
+        assetFile.write("\n")
+    assetFile.write("]\n")
+    assetFile.write("    }\n")
+    assetFile.write("}\n")
+    assetFile.close()
```

Comment:
This could share most of the code with 3dl_to_azasset.py, maybe rip it out to lut_helper?